### DISCUSSION TITLE
Add sorting by timestamp, date, etc to O2M fields

### DISCUSSION THREAD

**Discussion by NickCrews** [directus/directus#24700] [Feature Requests]
### Describe the Improvement

First of all, thank you so much for this package. I just started two days ago and have been very impressed so far.

I have a O2M relationship, and I want to show the list of linked items. The linked items have a `datetime` field, and I want to order the shown items by that. However, only [`['integer', 'bigInteger', 'float', 'decimal']`  are allowed to be used to sort](https://github.com/directus/directus/blob/b232e181564d091920cb6505812b0d72a394c713/app/src/modules/settings/routes/data-model/field-detail/field-detail-advanced/field-detail-advanced-relationship-o2m.vue#L66). Is there a reason for this? If so I would love to add a comment there. If this is possible, I can submit a PR!

See here, I want to be able to select `donation__date`:
<img width="855" alt="image" src="https://github.com/directus/directus/assets/10820686/4db0b6ca-60c0-464c-bb7f-83e67e394080">

**Comment by DanielBiegler** [comment]
Valid point, ordering by dates is useful! I wanted to find out why we added this so I followed some commits. This got introduced in #16379 because

> fix a minor bug caused by https://github.com/directus/directus/pull/14303 where you would not have been able to select a field with the name collection or item in the o2m sortField dropdown

So I went to #14303 which is fix for #14290 

> Re-implements the fix done here https://github.com/directus/directus/pull/11444 and here https://github.com/directus/directus/pull/12463 which were reverted in https://github.com/directus/directus/pull/13014 because the fix was also preventing relational fields to be selected as foreign keys for o2m/m2o.

So one reason it seems, with my very short investigation, is the following:

> When selecting a "sort field" that field will be used to both sort by and to save the sort order as it needs to save the sort order somewhere when updated.

So this sorting-by-dates might be a little more involved than just adding the type to the allow list. Might need modifications and checks to prevent reordering. A PR would be lovely :heart: and we'd appreciate it, but you probably have to dive deeper to make sure youre not breaking existing functionality and add new UI. :smile:

**Comment by bryantgillespie** [comment]
Running into this as well.

Would be super handy to be able to sort the related collection by a date_field because these are showing out of order for some reason.

![ScreenShot 2023-12-18 at 20 15 39@2x](https://github.com/directus/directus/assets/23302570/b06e34db-24e3-4a2c-bf6b-234ef06bc498)

**Comment by rrrnld** [comment]
I second this! Sort order and more field types would both be super helpful. Ideally this would be just as flexible as the `sort` parameter of the rest api, that's super nice to use.

**Comment by ertanar** [comment]
It would really be beneficial if one would be able to set a **Date/Timestamp** as the **Sort Field** of a **Collection**

**Comment by yasienkashef** [comment]
Upvoting this too. Its really unfortunate that you can only sort by integer values. In addition to that, in the table view of the collection (Layout of the collection) if I click on a column and want to sort it ascending or descending, it always sorts by id. It would be great to choose the sorting field like name or title.
